### DISCUSSION TITLE
aws-sdk-s3 1.117.0 returns wrong url when using _url() method when using Minio

### DISCUSSION THREAD

**Discussion by majksner** [shrinerb/shrine#602] [Q&A] [answered]
Hi There,

Before opening an issue, I wanted to check if anyone else has experienced an issue when using Minio locally.

After updating `aws-sdk-s3` gem from 1.116.0 to 1.117.0 _url() method returns wrong URL.

`aws-sdk-s3 - 1.116.0` returns http://127.0.0.1:9000/
`aws-sdk-s3 - 1.117.0` returns http://127.0.0.1:443/

```ruby
require('shrine/storage/s3')
storage_options = {
    bucket: ENV.fetch('BUCKET', nil),
    access_key_id: Rails.configuration.aws_access_key_id,
    secret_access_key: Rails.configuration.aws_secret_access_key,
    region: 'us-east-1',
    force_path_style: true,
    endpoint: 'http://127.0.0.1:9000'
  }
  Shrine.storages = {
    cache: Shrine::Storage::S3.new(prefix: 'cache', **storage_options),
    store: Shrine::Storage::S3.new(**storage_options)
  }

```

**Comment by janko** [comment]
Seems like a regression in aws-sdk-s3. Have you tried reporting it there, ideally with a minimal self-contained example without Shrine?

**Comment by majksner** [comment, marked as answer]
Definitely an issue with aws-sdk-s3. They confirmed it a regression. https://github.com/aws/aws-sdk-ruby/issues/2776
